### PR TITLE
[Docs] Translate model analysis document to English

### PR DESCRIPTION
**Pull request by enkilee**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

add model analysis english doc

## Modification

add model analysis english doc

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/common_usage/mode_analysis.md [hunk 1]
```diff
@@ -0,0 +1,740 @@
+# Count the calculation of Model and Parameter
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:1]:
```suggestion
# Calculate the FLOPs and Parameters of Model
```

File: docs/en/common_usage/mode_analysis.md [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+    def forward(self, imgs, labels=None, mode='tensor'):
+        x = self.resnet(imgs)
+        if mode == 'loss':
+            return {'loss': F.cross_entropy(x, labels)}
+        elif mode == 'predict':
+            return x, labels
+        elif mode == 'tensor':
+            return x
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:23]: Need 2 blank indentations

File: docs/en/common_usage/mode_analysis.md [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+
+2. Count the number of calculation and parameter
+
+```python
+from mmengine.analysis import get_model_complexity_info
+
+input_shape = (3, 224, 224)
+model = MMResNet50()
+analysis_results = get_model_complexity_info(model, input_shape)
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:33]: Need 2 blank indentations

File: docs/en/common_usage/mode_analysis.md [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+
+- Using table form to show
+
+```python
+print(analysis_results['out_table'])
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:39]: Need 2 blank indentations

File: docs/en/common_usage/mode_analysis.md [hunk 5]
```diff
@@ -0,0 +1,740 @@
...
+
+<details>
+    <summary>Click to expand</summary>
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:42]:
```suggestion
  <details>
  <summary>Click to expand</summary>
```

File: docs/en/common_usage/mode_analysis.md [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+|    layer4.2.conv2      |    2.359M            |    0.116G  |    25.088K   |
+|    layer4.2.bn2        |    1.024K            |    0.125M  |    0         |
+|    layer4.2.conv3      |    1.049M            |    51.38M  |    0.1M      |
+|    layer4.2.bn3        |    4.096K            |    0.502M  |    0         |
+|  fc                    |  2.049M              |  2.048M    |  1K          |
+|   fc.weight            |   (1000, 2048)       |            |              |
+|   fc.bias              |   (1000,)            |            |              |
+|  avgpool               |                      |  0.1M      |  0           |
++------------------------+----------------------+------------+--------------+
+```
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:179]: Need 2 blank indentations

File: docs/en/common_usage/mode_analysis.md [hunk 7]
```diff
@@ -0,0 +1,740 @@
...
+</details>
+
+- Using Model structure to show
+
+```python
+print(analysis_results['out_arch'])
+```
+
+<details>
+    <summary>点击展开</summary>
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:190]:
```suggestion
    <summary>Click to expand</summary>
```

File: docs/en/common_usage/mode_analysis.md [hunk 8]
```diff
@@ -0,0 +1,740 @@
...
+```
+
+- Total number of Parameters
+
+```python
+print("Model Parameters:{}".format(analysis_results['params_str']))
+# Model Parameters:25.557M
+```
+
+For the definition of model calculation and parameter and more usage, please refer[Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:740]:
```suggestion
For the definition of FLOPs and Parameters of model and more usage, please refer [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```

**Comment by zhouzaida** [comment]
Please refer to #939 to update the `docs/en/index.rst`

File: docs/en/common_usage/mode_analysis.md (hunk 2 above)
Comment (enkilee) [reply docs/en/common_usage/mode_analysis.md]: mdformat.................................................................Failed
- hook id: mdformat
- files were modified by this hook

if using 2 blank indentation, pre-commit errors. i find that some docs using 3 blank indentation. so i used

Comment (zhouzaida) [reply docs/en/common_usage/mode_analysis.md]: OK

File: docs/en/common_usage/mode_analysis.md [hunk 9]
```diff
@@ -0,0 +1,740 @@
...
+           x = self.resnet(imgs)
+           if mode == 'loss':
+               return {'loss': F.cross_entropy(x, labels)}
+           elif mode == 'predict':
+               return x, labels
+           elif mode == 'tensor':
+               return x
+   ```
+
+2. Count the number of calculation and parameter
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:25]:
```suggestion
2. Calculate the FLOPs and Parameters
```

File: docs/en/common_usage/mode_analysis.md [hunk 10]
```diff
@@ -0,0 +1,740 @@
...
+
+   ```python
+   from mmengine.analysis import get_model_complexity_info
+
+   input_shape = (3, 224, 224)
+   model = MMResNet50()
+   analysis_results = get_model_complexity_info(model, input_shape)
+   ```
+
+- Using table form to show
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:35]:
```suggestion
- Show in table form
```

File: docs/en/common_usage/mode_analysis.md [hunk 11]
```diff
@@ -0,0 +1,740 @@
...
+      in_features=2048, out_features=1000, bias=True
+      #params: 2.05M, #flops: 2.05M, #acts: 1K
+      )
+  )
+  )
+  ```
+
+  </details>
+
+- Total number of calculation
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:726]:
```suggestion
- Show FLOPs as a string
```

File: docs/en/common_usage/mode_analysis.md [hunk 12]
```diff
@@ -0,0 +1,740 @@
...
+  ```
+
+- Total number of Parameters
+
+  ```python
+  print("Model Parameters:{}".format(analysis_results['params_str']))
+  # Model Parameters:25.557M
+  ```
+
+For the definition of FLOPs and Parameters of model and more usage, please refer [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:740]:
```suggestion
For the definition of FLOPs and Parameters of model and more usage, please refer to [Model Complexity Analysis](../advanced_tutorials/model_analysis.md)
```

File: docs/en/common_usage/mode_analysis.md [hunk 13]
```diff
@@ -0,0 +1,740 @@
...
+  </details>
+
+- Total number of calculation
+
+  ```python
+  print("Model Flops:{}".format(analysis_results['flops_str']))
+  # Model Flops:4.145G
+  ```
+
+- Total number of Parameters
```
Comment (zhouzaida) [inline docs/en/common_usage/mode_analysis.md:733]:
```suggestion
- Show Parameters as a string
```

File: docs/en/index.rst [hunk 14]
```diff
@@ -25,6 +25,7 @@ You can switch between Chinese and English documents in the lower-left corner of
    common_usage/save_gpu_memory.md
    common_usage/debug_tricks.md
    common_usage/epoch_to_iter.md
+   common_usage/model_analysis.md
```
Comment (zhouzaida) [inline docs/en/index.rst:28]:
```suggestion
   common_usage/model_analysis.md
   common_usage/epoch_to_iter.md
```

**Comment by zhouzaida** [comment]
Hi, please rename `docs/en/common_usage/mode_analysis.md` to `docs/en/common_usage/model_analysis.md`.

**Review by zhouzaida — APPROVED** [review]

**Review by HAOCHENYE — COMMENTED** [review]
Thanks for your contributions! During the review, you've fixed some common issues that exist in both English and Chinese documents, it could be helpful to enhance the Chinese document at the same time :smile: !

File: docs/en/common_usage/model_analysis.md [hunk 15]
```diff
@@ -0,0 +1,740 @@
...
+  print("Model Flops:{}".format(analysis_results['flops_str']))
+  # Model Flops:4.145G
+  ```
+
+- Show Parameters as a string
+
+  ```python
+  print("Model Parameters:{}".format(analysis_results['params_str']))
+  # Model Parameters:25.557M
+  ```
```
[the remark this answers is not in the document]
Comment (enkilee) [reply docs/en/common_usage/model_analysis.md]: OK, changed it

**Review by zhouzaida — APPROVED** [review]
